sklego/meta.py

```diff
@@ -38,34 +38,113 @@ def transform(self, X):
+
+    res = np.zeros(len(group_sizes))
+    res[np.argmin(np.array(group_sizes) >= min_n_obs) - 1] = 1
+    return res
+
+
 class GroupedEstimator(BaseEstimator):
     """
     Construct an estimator per data group. Splits data by values of a
     single column and fits one estimator per such column.
 
     :param estimator: the model/pipeline to be applied per group
     :param groups: the column(s) of the matrix/dataframe to select as a grouping parameter set
-    :param use_fallback: weather or not to fall back to a general model in case
+    :param value_columns: Columns to use in the prediction. If None (default), use all non-grouping columns
+    :param use_fallback: whether or not to fall back to a general model in case
     the group parameter is not found during `.predict()`
+    :param shrinkage: How to perform shrinkage.
+                        None: No shrinkage (default)
+                        {"constant", "min_n_obs", "relative"} or a callable
+                        * constant: shrinked prediction for a level is weighted average of its prediction and its
+                                    parents prediction
+                        * min_n_obs: shrinked prediction is the prediction for the smallest group with at least
+                                     n observations in it
+                        * relative: each group-level is weight according to its size
+                        * function: a function that takes a list of group lengths and returns an array of the
+                                    same size with the weights for each group
+    :param **shrinkage_kwargs: keyword arguments to the shrinkage function
     """
-
-    def __init__(self, estimator, groups, use_fallback=True):
+    def __init__(self, estimator, groups, value_columns=None, use_fallback=True, shrinkage=None, **shrinkage_kwargs):
         self.estimator = estimator
         self.groups = groups
-        self.use_fallback = use_fallback
+        self.value_columns = value_columns
+        self.use_fallback = use_fallback  # Do we need this in case of shrinkage?
+        self.shrinkage = shrinkage
+        self.shrinkage_kwargs = shrinkage_kwargs
+
+    def __set_shrinkage_function(self):
+        if isinstance(self.shrinkage, str):
+            # Predefined shrinkage functions
+            shrink_options = {
+                "constant": constant_shrinkage,
+                "relative": relative_shrinkage,
+                "min_n_obs": min_n_obs_shrinkage,
+            }
+            try:
+                self.shrinkage_function = shrink_options.get(self.shrinkage)
+            except AttributeError:
+                raise ValueError(f"The specified shrinkage function {self.shrinkage} is not valid, "
+                                 f"choose from {list(shrink_options.keys())} or supply a callable.")
+        elif callable(self.shrinkage):
+            self.__check_shrinkage_func()
+            self.shrinkage_function = self.shrinkage_function
```

Comment:
what does this line do exactly?

Reply:
should this not be

```
self.shrinkage_function = self.shrinkage
````
?

Reply:
it seems worth a test to see if this does not go wrong.

Reply:
yep. added test. there's a problem here.

Reply:
Sharp, must have happened while refactoring. This test was already included, but caught another error. Your test failed for another reason: when only specifying one column, shrinkage is not possible. Your comment led to two improvements:
- Added the option to shrink to a global model
- Improved tests to look for specific errors, not just a `ValueError`